lib/msal-core/src/UserAgentApplication.ts

```diff
@@ -302,9 +314,12 @@ export class UserAgentApplication {
     loginRedirect(request?: AuthenticationParameters): void {
         // Throw error if callbacks are not set before redirect
         if (!this.redirectCallbacksSet) {
+            this.cacheStorage.setItem(INTERACTION_STATUS, RequestStatus.COMPLETED);
             throw ClientConfigurationError.createRedirectCallbacksNotSetError();
         }
-        this.acquireTokenInteractive(Constants.interactionTypeRedirect, true, request);
+        // append GUID to user set state  or set one for the user if null
+        const state = ServerRequestParameters.setState(request && request.state);
+        this.acquireTokenInteractive(Constants.interactionTypeRedirect, true, request,  null, null, state);
```

Comment:
can we clean up state on redirect cancelled?

Reply:
This fails too early for any cache state to be created.